charts/nri-metadata-injection/templates/deployment.yaml

```
@@ -1,46 +1,43 @@
-{{- if (include "nri-metadata-injection.cluster" .) }}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ template "nri-metadata-injection.fullname" . }}
+  name: {{ include "newrelic.common.naming.fullname" . }}
   namespace: {{ .Release.Namespace }}
   labels:
-    {{- include "nri-metadata-injection.labels" . | nindent 4 }}
+    {{- include "newrelic.common.labels" . | nindent 4 }}
 spec:
   replicas: {{ .Values.replicas }}
   selector:
     matchLabels:
-      {{- include "nri-metadata-injection.appLabel" . | nindent 6 }}
+      {{- include "newrelic.common.labels.selectorLabels" . | nindent 6 }}
   template:
     metadata:
 {{- if .Values.podAnnotations }}
       annotations:
 {{ toYaml .Values.podAnnotations | indent 8}}
 {{- end }}
       labels:
-        {{- include "nri-metadata-injection.labels" . | nindent 8 }}
+        {{- include "newrelic.common.labels" . | nindent 8 }}
     spec:
-      # Switching from dedicated service account to default one must be
-      # done explicitly, otherwise upgrade fails, trying to use old
-      # service account.
-      serviceAccount: default
-      {{- if .Values.podSecurityContext.enabled }}
+      {{- with include "newrelic.common.securityContext.pod" . }}
       securityContext:
-        runAsUser: {{ .Values.podSecurityContext.runAsUser }}
-        runAsGroup: {{ .Values.podSecurityContext.runAsGroup }}
-        fsGroup: {{ .Values.podSecurityContext.fsGroup }}
+        {{- . | nindent 8 -}}
       {{- end }}
       {{- if .Values.image.pullSecrets }}
       imagePullSecrets:
-{{ toYaml .Values.image.pullSecrets | indent 8 }}
+        {{- include "newrelic.common.images.renderPullSecrets" ( dict "pullSecrets" ( list .Values.image.pullSecrets ) "context" .) | nindent 8 -}}
       {{- end }}
       containers:
-      - name: {{ template "nri-metadata-injection.name" . }}
-        image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+      - name: {{ include "newrelic.common.naming.name" . }}
+        image: {{ include "newrelic.common.images.image" ( dict "imageRoot" .Values.image "context" .) }}
         imagePullPolicy: "{{ .Values.image.pullPolicy }}"
```

| quote

---

I think using `| quote` would give us values like `'"newrelic/k8s-metadata-injection:1.6.0"'`, it is really needed?

---

sorry for my bad context of comment , i mean this line , not sure if needed the quotes though , but i would remove the or use the |quote func
```suggestion
        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
```

---

i think we can remove them actually https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy

---

My bad, I thought you meant the line I changed! 😅

Anyway I think you are definitely right, I'm removing those quotes

---

Done: [54761b5](https://github.com/newrelic/k8s-metadata-injection/pull/163/commits/54761b54e5a3ce75dbc1be8d314161d76ddf64d8)